apps/core/lib/core/event_repo.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Core.EventRepo do
+  @type operation ::
+          :user_created
+          | :user_updated
+          | :user_deleted
+          | :post_created
+
+@spec save_event(operation, {pos_integer(), String.t(), any()}) :: :ok | {:error, any()}
+  def save_event(:user_created, {user_id, user_email, time}),
+    do: write_to_mnesia(user_id, :user_created, %{user_email: user_email}, time)
+
+  def save_event(:user_updated, {user_id, attrs, time}),
+    do: write_to_mnesia(user_id, :user_updated, attrs, time)
+
+  def save_event(:user_deleted, {user_id, time}),
+    do: write_to_mnesia(user_id, :user_deleted, %{}, time)
+
+  def save_event(:post_created, {user_id, post_id, post_body, time}),
+    do:
+      write_to_mnesia(
+        user_id,
+        :post_created,
+        %{user_id: user_id, post_id: post_id, post_body: post_body},
+        time
+      )
+
+  def save_event(_, _), do: {:error, :invalid_operation}
+
+  defp write_to_mnesia(user_id, operation, attrs, time) do
```

Comment:
I think `save_event` reshuffling arguments is a bit confusing. I would consider making it a very thin layer on top of `write_to_mnesia` like so:

```elixir
def save_event(user_id, :user_created, time, attrs) ...
def save_event(user_id, :user_updated, time, attrs) ...
def save_event(user_id, :user_deleted, time) ...
def save_event(user_id, :post_created, time, attrs) ...
```

and then at the call site e.g.:

```elixir
EventRepo.save_event(user.id, :user_updated, user.updated_at, attrs)
```

I also moved `time` to the third position in the list of arguments for consistency.

Reply:
Since this is just a thin layer for a well-defined list of supported events, I would make it even:

```elixir
def user_created(user_id, time, attrs)
```

or 

```elixir
def track_user_created(user_id, time, attrs)
```

Reply:
I really like @smt116 's solution